infra/modules/azure_naming_convention/README.md

```diff
@@ -1,5 +1,199 @@
 # DX - Common Variables Validation
 
+This module generates output values that follow the naming conventions used in DX. It takes the following inputs:
+
+- `prefix`: The prefix for the naming convention (e.g., io, dx, ...).
+- `env_short`: A short identifier for the environment (e.g., “d”, "p", "u").
+- `location`: The Azure region or location (e.g. italynorth, westeurope, ...).
+- `domain`: (Optional) The domain name to include in the naming convention.
+- `app_name`: The application name or identifier.
+- `instance_number`: The instance number that can be between 1 and 99.
+
+## Outputs
+
+The module generates output values based on whether the configuration targets a single environment or multiple environments.
+
+### Single Environment Outputs
+
+If the module is configured for a single environment, the following outputs are generated:
+
+- `prefix`: Combines prefix, env_short, location_short, optionally including the domain and app_name. The format is:
+`prefix-env_short-location_short-domain-app_name` (e.g. `dx-d-itn-test-eg` with domain as test and app_name as eg).
+- `suffix`: A map of instance_number values, represented as two-digit strings. Example:
+
+  ```hcl
+  {
+    "1": "01",
+    "2": "02",
+    ...
+  }
+  ```
+
+- `project`: A combination of prefix, env_short, and location_short values. The format is:
+`prefix-env_short-location_short`. Example: `dx-d-itn`.
+- `name`: A map where resource types serve as keys and the values are resource names.
+Resource names follow the format: `prefix-abbreviation-suffix`. The abbreviation is retrieved from the `local.resource_abbreviations` mapping.
+  Example:
+
+  ```hcl
+  {
+    "app_service" = {
+      "1" = "dx-d-itn-test-eg-app-01",
+      "2" = "dx-d-itn-test-eg-app-02",
+      ...
+    },
+    "blob_storage" = {
+      "1" = "dx-d-itn-test-eg-blob-01",
+      "2" = "dx-d-itn-test-eg-blob-02",
+      ...
+    },
+    "cosmos_db_nosql" = {
+      "1" = "dx-d-itn-test-eg-conmo-01",
+      "2" = "dx-d-itn-test-eg-conmo-02",
+      ...
+    },
+    "virtual_network" = {
+      "1" = "dx-d-itn-test-eg-vnet-01"
+      "2" = "dx-d-itn-test-eg-vnet-02",
+      ...
+    },
+    ...
+  }
+  ```
+
+### Multi-Environment Outputs
+
+For configurations with multiple environments, the following outputs are generated:
+
+- `prefixes`:  A list of all prefixes, where each combines prefix, env_short, location_short, optionally including domain and app_name. The format is:
+`[prefix-env_short-location_short-domain-app_name,..]` (e.g. `dx-d-itn-test-eg` if domain is test and app_name is eg).
+- `suffixes`: A map where each key is a prefix, and the value is a map of `instance_number` values as two-digit strings. Example:
+
+  ```hcl
+  {
+    "dx-d-itn-test-eg" = {
```

Comment:
why not just use the env_short name? I'll try to use one structure for both multi env and mono env. ie

```
{

  "d" = {
      "app_service" = {
        "1" = "dx-d-itn-test-eg-app-01",
        "2" = "dx-d-itn-test-eg-app-02",
        ...
      },
      "blob_storage" = {
        "1" = "dx-d-itn-test-eg-blob-01",
        "2" = "dx-d-itn-test-eg-blob-02",
        ...
      },
      "cosmos_db_nosql" = {
        "1" = "dx-d-itn-test-eg-conmo-01",
        "2" = "dx-d-itn-test-eg-conmo-02",
        ...
      },
      "virtual_network" = {
        "1" = "dx-d-itn-test-eg-vnet-01"
        "2" = "dx-d-itn-test-eg-vnet-02",
        ...
      },
      ...
    },

   "p" = { ... }

]


```

Reply:
ENV_SHORT is not unique, if there is multiple environments with same env_short will not work correctly

Reply:
In which case there are multiple environments with the same `env_short` names?

Reply:
Anyway, we assume that APP_PREFIX `dx-d-itn-test-app` is unique, but it can be computed by the naming module itself

Reply:
Never inside the modules, but as @lucacavallaro  and @BurnedMarshal had pointed out to us, it happened that they used different environments and therefore were forced to instantiate the module multiple times. This is why we opted for this multi-environment solution in the end.

The problem here would then be that in a single instance you can call all the environments you want, and since, for example, you can make multiple environments with only `app_name` that varies, or only the `env`, the key had to be made as a composite value.

If we find a way to simplify the key obviously it would be better, for now I have inserted the most effective form.

Reply:
do these "different environments" differentiate by `env_short`? or what else? in the first case `env_short` would be a unique key.